src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+## Custom Primary Keys
+
+By default, all DataStore models have an `id` field by default that is automatically populated on the client with a UUID v4, allowing DataStore to generate non-colliding globally unique identifiers in a scalable way.
+
+While UUIDs have desirable properties (they are large, non-sequential and opaque), there are times when a custom primary key is needed. For instance, to:
+
+
+* Have friendly/readable identifiers (surrogate/opaque vs. natural keys)
+* Define composite primary keys
+* Customize data partitioning to optimize for scale (especially important when planning to handle large amounts of data in short periods of time)
+* Selectively synchronize data to clients (e.g. by fields like `deviceId`, `userId` or similar)
+* Prioritize the sort order in which objects are returned by the sync queries
+* Make existing data consumable and syncable by DataStore
+
+### Schema
```

Comment:
This might not need its own heading. One thing we need to ensure is that these undescriptive headings don't end up polluting the search results in the docs search.

Reply:
It feels disjointed without a heading (first section is the "Summary" / "Intro", here we're starting to get into actual usage). Thoughts on what else we could call this?

Reply:
Wdyt of "Example schemas using custom primary keys" - the reason for the "custom primary keys" is just making sure the search works better. Same logic as for the remaining "creating records..." etc. comments below.

Reply:
Makes sense! Updated